data/basics/testing/nearzero.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2019-2025 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package testing
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+// NearZeros takes a sample in order to retrieve its type. It returns a slice of
+// the same type in which each element of the slice is the same type as the
+// sample, but exactly one field (or sub-field) is set to a non-zero value. It
+// returns one example for every sub-field.
+func NearZeros(t *testing.T, sample any) []any {
```

Comment:
maybe add a simple random test creating a struct with reflect with a few fields of random type and make sure this function returns that exact number of element in a slice?

Reply:
You mean a test for `NearZeros` itself? I agree, I'll write some.